state/allwatcher.go

```diff
@@ -347,30 +357,32 @@ func (u *backingUnit) updated(st *State, store *multiwatcherStore, id string) er
 	if u.CharmURL != nil {
 		info.CharmURL = u.CharmURL.String()
 	}
+
+	// fetch the associated unit to get possible updated status.
+	unit, err := st.Unit(u.Name)
+	if err != nil {
+		return errors.Annotatef(err, "cannot get unit %q", u.Name)
+	}
+
 	oldInfo := store.Get(info.EntityId())
 	if oldInfo == nil {
-		logger.Debugf("new unit %q added to backing state", u.Name)
-		// We're adding the entry for the first time,
-		// so fetch the associated unit status and opened ports.
-		unitStatus, agentStatus, err := unitAndAgentStatus(st, u.Name)
+		unitStatusResult, err := unit.Status()
```

Comment:
can we keep the debug message and comment
also, no need to split the func calls, use the original method